src/components/KTreeList/KTreeList.vue

```diff
@@ -0,0 +1,419 @@
+      selectedItem.selected = true
+
+      if (selectedItem.children?.length) {
+        handleSelection(itemToSelect, selectedItem.children)
+      }
+    }
+    // deselect previously selected item
+    const unselectedItems = internalList.value.filter((item: TreeListItem) => item.id !== itemToSelect.id)
+    unselectedItems.forEach((item: TreeListItem) => {
+      item.selected = false
+      if (item.children?.length) {
+        handleSelection(itemToSelect, item.children)
+      }
+    })
+  } else { // recurse through children
+    // select the item
+    const selectedItem = list.filter((item: TreeListItem) => item.id === itemToSelect.id)?.[0]
+    if (selectedItem) {
+      selectedItem.selected = true
+
+      if (selectedItem.children?.length) {
+        handleSelection(itemToSelect, selectedItem.children)
+      }
+    }
+    // deselect previously selected item
+    const unselectedItems = list.filter((item: TreeListItem) => item.id !== itemToSelect.id)
+    unselectedItems.forEach((item: TreeListItem) => {
+      item.selected = false
+      if (item.children?.length) {
+        handleSelection(itemToSelect, item.children)
+      }
+    })
+  }
+  emit('selected', itemToSelect)
+}
+
+const handleSetParent = (item: TreeListItem, parent: TreeListItem) => {
+  let originalParent:TreeListItem
+  const selectedItem = internalList.value.filter((anItem: TreeListItem) => {
+    if (anItem.children?.length) {
+      const arr = anItem.children.filter((aChild: TreeListItem) => aChild.id === item.id)
+      if (arr.length) {
+        originalParent = anItem
+        return arr[0]
+      }
+    }
+    return anItem.id === item.id
+  })?.[0]
+
+  if (selectedItem) {
+    const parentItem = internalList.value.filter((anItem: TreeListItem) => anItem.id === parent.id)?.[0]
+    if (parentItem) {
+      // set new parent
+      if (!parentItem.children) {
+        parentItem.children = []
+      }
+      parentItem.children.push(selectedItem)
+    }
+  }
+
+  // TS not figuring out originalParent may have been set
+  // @ts-ignore
+  if (originalParent) {
+    // remove the item from the original parent
+    if (!originalParent.children) {
+      originalParent.children = []
+    }
+    originalParent.children = originalParent.children.filter((child: TreeListItem) => child.id !== item.id)
+  }
+}
+
+const handleChangeEvent = () => {
+  if (props.parentId) {
+    emit('child-change', {
+      parent: props.parentId,
+      children: internalList.value,
+    })
+  } else {
+    emit('change', {
+      items: internalList.value,
+    })
+  }
+}
+
+const maxLevelReached = computed(() => {
+  return props.level > props.maxLevel
+})
+
+/**
+     * Recursive check to get the maximum depth of an object.
+     * Documented here: https://stackoverflow.com/a/48505969
+     */
+const getMaximumDepth = ({ children = [] }): number => {
+  return children.length === 0 ? 0 : 1 + Math.max(...children.map(getMaximumDepth))
+}
+
+const checkMove = (target: any) => {
```

Comment:
I think this one is right?
```suggestion
const checkMove = (target: any): boolean => {
```